src/RadixDlt.NetworkGateway.PostgresIntegration/Services/PendingTransactionResponseAggregator.cs

```diff
@@ -0,0 +1,292 @@
+            payloadHash: pendingTransactionSummary.PayloadHash,
+            status: legacyStatus,
+            payloadStatus: payloadStatus,
+            payloadStatusDescription: GetPayloadStatusDescription(payloadStatus),
+            errorMessage: initialRejectionReason,
+            latestErrorMessage: latestRejectionReason == initialRejectionReason ? null : latestRejectionReason,
+            handlingStatus: handlingStatus,
+            handlingStatusReason: pendingTransactionSummary.HandlingStatusReason,
+            submissionError: pendingTransactionSummary.LastSubmissionError
+        ));
+    }
+
+    internal GatewayApiSdk.Model.TransactionStatusResponse IntoResponse()
+    {
+        var (intentStatus, errorMessage) = _committedIntentStatus != null
+            ? (_committedIntentStatus.Value, _committedErrorMessage)
+            : (
+                _mostAccuratePendingTransactionIntentLedgerStatus switch
+                {
+                    PendingTransactionIntentLedgerStatus.Unknown => GatewayApiSdk.Model.TransactionIntentStatus.Unknown,
+                    PendingTransactionIntentLedgerStatus.Committed => GatewayApiSdk.Model.TransactionIntentStatus.CommitPendingOutcomeUnknown,
+                    PendingTransactionIntentLedgerStatus.CommitPending => GatewayApiSdk.Model.TransactionIntentStatus.CommitPendingOutcomeUnknown,
+                    PendingTransactionIntentLedgerStatus.PermanentRejection => GatewayApiSdk.Model.TransactionIntentStatus.PermanentlyRejected,
+                    PendingTransactionIntentLedgerStatus.PossibleToCommit => GatewayApiSdk.Model.TransactionIntentStatus.Pending,
+                    PendingTransactionIntentLedgerStatus.LikelyButNotCertainRejection => GatewayApiSdk.Model.TransactionIntentStatus.LikelyButNotCertainRejection,
+                },
+                _rejectionReasonForMostAccurateIntentLedgerStatus
+            );
+
+        var legacyIntentStatus = intentStatus switch
+        {
+            GatewayApiSdk.Model.TransactionIntentStatus.Unknown => GatewayApiSdk.Model.TransactionStatus.Unknown,
+            GatewayApiSdk.Model.TransactionIntentStatus.CommittedSuccess => GatewayApiSdk.Model.TransactionStatus.CommittedSuccess,
+            GatewayApiSdk.Model.TransactionIntentStatus.CommittedFailure => GatewayApiSdk.Model.TransactionStatus.CommittedFailure,
+            GatewayApiSdk.Model.TransactionIntentStatus.CommitPendingOutcomeUnknown => GatewayApiSdk.Model.TransactionStatus.Pending,
+            GatewayApiSdk.Model.TransactionIntentStatus.PermanentlyRejected => GatewayApiSdk.Model.TransactionStatus.Rejected,
+            GatewayApiSdk.Model.TransactionIntentStatus.LikelyButNotCertainRejection => GatewayApiSdk.Model.TransactionStatus.Pending,
+            GatewayApiSdk.Model.TransactionIntentStatus.Pending => GatewayApiSdk.Model.TransactionStatus.Pending,
+        };
+
+        var statusesToReturnPermanentlyRejectsAtEpoch = new[]
```

Comment:
This should be a static readonly value, not a variable.

Reply:
well it's the only place where it's used. separating it from usage will only make you jump between two places and harder to read the code. I see no benefit from doing that.

Reply:
This is static, constant value, not a variable. Feel free to define it as static within the narrowest possible scope. We don't have to allocate it on every single usage.